content/en/platform/corda/5.0/developing-applications/api/ledger/utxo-ledger/vault-queries.md

```diff
@@ -0,0 +1,561 @@
+
+
+This defines whether or not to keep the given element (`row`) from the result set. Elements returning true are kept and the rest are discarded.
+
+In this example, keep the elements that have “Alice” in their participant list. This filter would look like this:
+
+{{< tabs name="tabs-4" >}}
+{{% tab name="Kotlin" %}}
+
+```kotlin
+class DummyCustomQueryFilter : VaultNamedQueryStateAndRefFilter<TestState> {
+    override fun filter(data: StateAndRef<TestUtxoState>, parameters: MutableMap<String, Any>): Boolean {
+        return data.state.contractState.participantNames.contains("Alice")
+    }
+}
+```
+
+{{% /tab %}}
+{{% tab name="Java" %}}
+
+```java
+class DummyCustomQueryFilter implements VaultNamedQueryStateAndRefFilter<TestUtxoState> {
+
+    @NotNull
+    @Override
+    public Boolean filter(@NotNull StateAndRef<TestUtxoState> data, @NotNull Map<String, Object> parameters) {
+        return data.getState().getContractState().getParticipantNames().contains("Alice");
+    }
+}
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+#### Transforming
+
+To create a transformer class make sure to only keep the transaction IDs of each record. Transformer classes need to implement the `VaultNamedQueryStateAndRefTransformer<T, R>` interface. The `<T>` is the type of results returned from the database, in this case `TestState` and `<R>` is the type to transform the results into and to have transaction IDs which are `Strings`.
+
+This interface has one function:
+
+```java
+@NotNull`
+`R transform(@NotNull T data, @NotNull Map<String, Object> parameters);
+```
+
+This defines how each record (“row”) will be transformed (mapped):
+
+{{< tabs name="tabs-5" >}}
+{{% tab name="Kotlin" %}}
+
+```kotlin
+class DummyCustomQueryTransformer : VaultNamedQueryStateAndRefTransformer<TestState, String> {
+    override fun transform(data: StateAndRef<TestState>, parameters: MutableMap<String, Any>): String {
+        return data.ref.transactionId.toString()
+    }
+}
+```
+
+{{% /tab %}}
+{{% tab name="Java" %}}
+
+```java
+
+class DummyCustomQueryMapper implements VaultNamedQueryStateAndRefTransformer<TestUtxoState, String> {
+    @NotNull
+    @Override
+    public String transform(@NotNull StateAndRef<TestUtxoState> data, @NotNull Map<String, Object> parameters) {
+        return data.getRef().getTransactionId().toString();
+    }
+}
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+This transforms each element to a `String` object which is the given state’s transaction ID.
+
+#### Collecting
+
+Collecting is used to collect results set into one single integer.
+
+For this, implement the `net.corda.v5.ledger.utxo.query.VaultNamedQueryCollector<R, T>` interface. The `<R>` parameter is the type of the original result set (in this case `String` because of transformation) and `<T>` is the type  collected into (in our case this is an `Int`).
```

Comment:
```suggestion
For this, implement the `net.corda.v5.ledger.utxo.query.VaultNamedQueryCollector<R, T>` interface. The `<R>` parameter is the type of the original result set (in this case `String` because of transformation) and `<T>` is the type collected into (in this can an `Int`).
```